### PR TITLE
Fix UnicodeDecodeError in C file reading

### LINKED ISSUE #72: UnicodeDecodeError when reading C file due to 'charmap' codec
When attempting to read a C file , I am encountering a UnicodeDecodeError with the message "'charmap' codec can't decode byte 0x90 in position 203280: character maps to <undefined>". This issue seems to be related to the decoding process when reading the file.

### PR DESCRIPTION
**Pull request by kirangda**
closes #72

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
Could you add an entry to the changelog ?

**Comment by kirangda** [comment]
> Could you add an entry to the changelog ?

Done.

File: CHANGES [hunk 1]
```diff
@@ -1,6 +1,11 @@
 PyCLibrary Changelog
 ====================
 
+0.2.2 - 01/12/2023
```
Comment (MatthieuDartiailh) [inline CHANGES:4]:
```suggestion
0.2.2 - unreleased
```

**Comment by MatthieuDartiailh** [comment]
Thanks (I simply changed the date since I am unlikely to release today).

**Pull request merged**